src/V3Sched.cpp

```diff
@@ -135,24 +170,21 @@ void splitCheck(AstCFunc* ofuncp) {
     while (tempp->stmtsp()) {
         AstNode* const itemp = tempp->stmtsp()->unlinkFrBack();
         const int stmts = itemp->nodeCount();
-        if (!funcp || (func_stmts + stmts) > v3Global.opt.outputSplitCFuncs()) {
-            // Make a new function
-            funcp = new AstCFunc{ofuncp->fileline(), ofuncp->name() + "__" + cvtToStr(funcnum++),
-                                 ofuncp->scopep()};
-            funcp->dontCombine(true);
-            funcp->isStatic(false);
-            funcp->isLoose(true);
-            funcp->slow(ofuncp->slow());
-            ofuncp->scopep()->addBlocksp(funcp);
-            //
-            AstCCall* const callp = new AstCCall{funcp->fileline(), funcp};
-            callp->dtypeSetVoid();
-            ofuncp->addStmtsp(callp->makeStmt());
-            func_stmts = 0;
+
+        if ((func_stmts + stmts) > v3Global.opt.outputSplitCFuncs()) {
+            finishSubFuncp(funcp);
+            funcp = createNewSubFuncp();
         }
+
         funcp->addStmtsp(itemp);
         func_stmts += stmts;
+
+        if (is_ofuncp_coroutine && !is_coroutine)
+            itemp->foreach([&is_coroutine](const AstNode* nodep) {
+                if (VN_IS(nodep, CAwait)) is_coroutine = true;
+            });
```

Comment:
I think this could be moved to `finishSubFuncp`? Just search the created function's statements. Then we wouldn't need the `is_coroutine` bool. Also, I suggest using `exists` here, i.e.
```cpp
funcp->exists([](const AstCAwait*) { return true; })
```

Reply:
Done.